pkg/controllers/management/kontainerdrivermetadata/data.go

```diff
@@ -206,15 +206,22 @@ func (md *MetadataController) createOrUpdateSystemImageCRD(k8sVersion string, sy
 			return err
 		}
 		return nil
-
 	}
-	if reflect.DeepEqual(sysImage.SystemImages, systemImages) && reflect.DeepEqual(sysImage.Labels, pluginsMap) {
+	dataEqual := reflect.DeepEqual(sysImage.SystemImages, systemImages)
```

Comment:
are these map or list comparisons (on my phone so hard to tell). If they are lists, deep equals is dangerous bc it takes into account order but we don’t really care about order of labels (do we?)

Reply:
I’m not sure on this one, just my reaction. So if you feel deepequals is safe in this (and the other below cases) respond with justification and we can move on

Reply:
SystemImages is struct comparison with all fields string, so thought it should be ok? 

For labels, I use custom logic https://github.com/rancher/rancher/blob/b4ba60ce4408b75dda0f2f33a4ab5b0227d09c55/pkg/controllers/management/kontainerdrivermetadata/data.go#L688

Reply:
Ok